deepmd/train/trainer.py

```diff
@@ -321,15 +338,27 @@ def _build_lr(self):
 
     def _build_network(self, data):        
         self.place_holders = {}
-        data_dict = data.get_data_dict()
-        for kk in data_dict.keys():
-            if kk == 'type':
-                continue
-            prec = GLOBAL_TF_FLOAT_PRECISION
-            if data_dict[kk]['high_prec'] :
-                prec = GLOBAL_ENER_FLOAT_PRECISION
-            self.place_holders[kk] = tf.placeholder(prec, [None], name = 't_' + kk)
-            self.place_holders['find_'+kk] = tf.placeholder(tf.float32, name = 't_find_' + kk)
+        if self.is_compress :
+            for kk in ['coord', 'box']:
+                self.place_holders[kk] = tf.placeholder(GLOBAL_TF_FLOAT_PRECISION, [None], 't_' + kk)
+            for kk in data_requirement.keys():
+                if kk == 'type':
+                    continue
+                prec = GLOBAL_TF_FLOAT_PRECISION
+                if data_requirement[kk]['high_prec'] :
+                    prec = GLOBAL_ENER_FLOAT_PRECISION
+                self.place_holders[kk] = tf.placeholder(prec, [None], name = 't_' + kk)
+                self.place_holders['find_' + kk] = tf.placeholder(tf.float32, name = 't_find_' + kk)
+        else :
+            data_dict = data.get_data_dict()
+            for kk in data_dict.keys():
+                if kk == 'type':
+                    continue
+                prec = GLOBAL_TF_FLOAT_PRECISION
+                if data_dict[kk]['high_prec'] :
+                    prec = GLOBAL_ENER_FLOAT_PRECISION
+                self.place_holders[kk] = tf.placeholder(prec, [None], name = 't_' + kk)
+                self.place_holders['find_' + kk] = tf.placeholder(tf.float32, name = 't_find_' + kk)
```

Comment:
DANGEROUS: copied code from line 345:352

Reply:
Any idea to get the place_holders without accessing to the training data?

Reply:
Write a method for generating place_holders